app/controllers/application_controller.rb

```diff
@@ -5,6 +5,139 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, 'this_is_a_secret'
   end
 
+  helpers do 
+    def current_user
+      User.find(session[:user_id])
+     end
+ 
+     def logged_in?
+       !!session[:user_id]
+     end
+  end
+
+  get '/' do 
+    erb :'/index'
+  end
+
+  get '/signup' do
+    if logged_in?
+      redirect '/tweets'
+    else
+    erb :'users/create_user'
+    end
+  end
+
+  post '/signup' do 
+    if params[:username] == "" || params[:email] == "" || params[:password] == ""
+      redirect '/signup'
+    else
+    @user = User.create(params)
+    session[:user_id] = @user.id 
+    redirect '/tweets'
+    end
+  end
+
+  get '/login' do 
+    if logged_in?
+      redirect '/tweets'
+    else
+    erb :'users/login'
+    end
+  end
+
+  post '/login' do 
+    @user = User.find_by(username: params[:username])
+    if @user && @user.authenticate(params[:password])
+    session[:user_id] = @user.id
+      redirect '/tweets'
+    else
+      redirect 'users/login'
+    end
+  end
+
+  get '/tweets' do 
+    if logged_in?
+    @tweets = Tweet.all
+    erb :'tweets/tweets'
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/tweets/new' do 
+    if logged_in?
+      erb :'tweets/create_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+  post '/tweets' do 
+    if params[:content].empty?
+      redirect '/tweets/new'
+    else
+      @tweet = Tweet.create(content: params[:content], user: current_user)
```

Comment:
Another way to do this is to make use of the `has_many` relationship on User:
```
@tweet = current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.